File: DatabindingWithRecyclerView/app/src/main/java/com/enpassio/databindingwithrecyclerview/ui/DetailsFragment.java
package com.enpassio.databindingwithrecyclerview.ui;


import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.enpassio.databindingwithrecyclerview.R;
import com.enpassio.databindingwithrecyclerview.databinding.FragmentDetailsBinding;
import com.enpassio.databindingwithrecyclerview.model.Product;

import static com.enpassio.databindingwithrecyclerview.ui.ProductListFragment.PRODUCT_KEY;

public class DetailsFragment extends Fragment {

    public DetailsFragment() {
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        FragmentDetailsBinding binding = DataBindingUtil.inflate(
                inflater, R.layout.fragment_details, container, false);

        //These are for making up button work. Not related to databinding.
        ((AppCompatActivity) getActivity()).setSupportActionBar(binding.toolbar);
        setHasOptionsMenu(true);

        Bundle bundle = getArguments();
        if (bundle != null) {
            /*Once we get the chosen product from the bundle,
            we pass it to the binding implementation*/
            Product chosenProduct = bundle.getParcelable(PRODUCT_KEY);
            binding.setProduct(chosenProduct);
        }

        return binding.getRoot();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        //This is for making up button in the toolbar behave like back button
        if (item.getItemId() == android.R.id.home) {
            FragmentManager fm = getFragmentManager();
            if (fm != null) {
                fm.popBackStack();
            }
        }
        return super.onOptionsItemSelected(item);
    }
}

File: DatabindingWithRecyclerViewKotlin/app/src/main/java/com/enpassio/databindingwithrecyclerviewkotlin/ui/DetailsFragment.kt
package com.enpassio.databindingwithrecyclerviewkotlin.ui

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import com.enpassio.databindingwithrecyclerviewkotlin.R
import com.enpassio.databindingwithrecyclerviewkotlin.databinding.FragmentDetailsBinding
import com.enpassio.databindingwithrecyclerviewkotlin.model.Product
import com.enpassio.databindingwithrecyclerviewkotlin.ui.ProductListFragment.Companion.PRODUCT_KEY


/**
 * Created by Greta Grigutė on 2018-12-23.
 */


class DetailsFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding = DataBindingUtil.inflate<FragmentDetailsBinding>(
                inflater, R.layout.fragment_details, container, false)

        //These are for making up button work. Not related to databinding.
        (activity as AppCompatActivity).setSupportActionBar(binding.toolbar)
        setHasOptionsMenu(true)

        val bundle = arguments
        if (bundle != null) {
            /*Once we get the chosen product from the bundle,
            we pass it to the binding implementation*/
            val chosenProduct = bundle.getParcelable<Product>(PRODUCT_KEY)
            binding.product = chosenProduct
        }

        return binding.root
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        //This is for making up button in the toolbar behave like back button
        if (item!!.itemId == android.R.id.home) {
            fragmentManager!!.popBackStack()
        }
        return super.onOptionsItemSelected(item)
    }
}